ark/analysis/spatial_analysis.py

```diff
@@ -536,9 +546,17 @@ def create_neighborhood_matrix(all_data, dist_mat_dir, included_fovs=None, distl
         cell_neighbor_freqs.loc[current_fov_neighborhood_data.index, fov_cluster_names] = freqs
 
     # Remove cells that have no neighbors within the distlim
-    keep_cells = cell_neighbor_counts.drop([fov_col, cell_label_col], axis=1).sum(axis=1) != 0
-    cell_neighbor_counts = cell_neighbor_counts.loc[keep_cells].reset_index(drop=True)
-    cell_neighbor_freqs = cell_neighbor_freqs.loc[keep_cells].reset_index(drop=True)
+    if not mixing:
+        total_cell_count = cell_neighbor_counts.shape[0]
+        keep_cells = cell_neighbor_counts.drop([fov_col, cell_label_col], axis=1).sum(axis=1) != 0
+        cell_neighbor_counts = cell_neighbor_counts.loc[keep_cells].reset_index(drop=True)
+        cell_neighbor_freqs = cell_neighbor_freqs.loc[keep_cells].reset_index(drop=True)
+        # issue warning if more than 5% of cells are dropped
```

Comment:
Do we need separate logic for mixing and non-mixing? This means people can't reuse the same neighborhood matrix for the different analyses, right? What's the reason we can't have the same matrix for both?